src/App.tsx

```diff
@@ -14,11 +15,131 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+
+type State = {
+  goods: string[],
+  visibilityOfGoods: boolean,
+  isReversed: boolean,
+  sortBy: string,
+};
+
+export class App extends React.Component<Props, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    visibilityOfGoods: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  start = () => {
+    this.setState(() => ({
+      visibilityOfGoods: true,
+    }));
+  };
+
+  reverse = () => (
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }))
+  );
+
+  sortGoods = (sortBy: string) => (
+    this.setState(() => ({
+      sortBy,
+    }))
+  );
+
+  reset = () => {
+    this.setState(() => ({
+      sortBy: '',
+      isReversed: false,
+    }));
+  };
+
+  render(): React.ReactNode {
+    const { visibilityOfGoods, isReversed, sortBy } = this.state;
+    const goods = [...this.state.goods];
+
+    if (sortBy.length !== 0) {
+      switch (sortBy) {
+        case 'by length':
+          goods.sort((a, b) => a.length - b.length);
+          break;
+
+        case 'alphabetically':
+          goods.sort((a, b) => a.localeCompare(b));
+          break;
+
+        default:
+          return 0;
+      }
+    }
+
+    if (isReversed) {
+      goods.reverse();
+    }
```

Comment:
Create separate function for this logic, e. g. `getVisibleGoods()`. Method `render()` should be as clear as possible, it should render, but not implement some specific logic.